IN-CORE/pyincore
# 337 refactor building damage analysis to support multiple hazards

I made a few hacks mainly around the demand types
1. Since there is no way to tell which demand type belongs what hazard, I have to hardcode a list of allowed hazard type for tsunami and earthquake
2. Need to record the mapping of original demand type vs adjusted (hazard service) returned hazard type. Original code just assume the **same length and order** (read the hval_dict[j] part)

E.g. 
- 1.04 Sec Sa need to be converted to 1.04 SA before sending to the services (payload); but then when it's returned it further becomes 1.0 SA
- But the original equation in the fragility curve is using 1.04 Sec SA
- Hence I invented a dictionary to record that

Not proud of that code I wrote but I can't think of better ways without fundamentally change the pyincore as well as services code (demand type data model)


Anyway it works; to test:
- run the test_buildingdamage.py and compare with old results to see if backward compatible
- check the `seaside_multihazard_` result
